core/src/test/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtilTest.java

```diff
@@ -116,6 +116,12 @@ void testDecodePath(boolean isPathParam) throws Exception {
         } else {
             assertThat(decodeFunc.apply("/%2F")).isEqualTo("/%2F");
         }
+
+        // Tests for '+' decoding into ' '
+        assertThat(decodeFunc.apply("/foo+bar")).isEqualTo("/foo bar"); // '+' should be decoded into space
```

Comment:
+ should be decoded only when it's used as a part of a query string. `/foo+bar` stays same while `/foo?bar+baz` is decoded into `/foo?bar baz`.